### ISSUE TITLE
Threads in which not mentioned or sent messages are showing up in the LHN for user

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#28253]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1.Join offshore social room which had bunch of threads

## Expected Result:
LHN should show the threads for which message is sent

## Actual Result:
 LHN shows all the threads shared with me. It happens only for threads that existed before I joined

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 
**Reproducible in staging?:** needs reproduction
**Reproducible in production?:** needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:** @iwiznia 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1695638096485759

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01db41e6a729ab5619</li>
        <li>Upwork Job ID: 1716799730931810304</li>
        <li>Last Price Increase: 2023-10-24</li>
    </ul>
</details>

**Comment by srikarparsi** [comment]
Hey, sorry, wasn't able to prioritize this because I wanted to finish the core threads implementation before tackling follow up issues. Will take a look into this on Wednesday.

**Comment by srikarparsi** [comment]
Sounds good, I'm planning on investigating this on Monday since I'm a little busy this week

**Comment by srikarparsi** [comment]
I tried reproducing this today but couldn't. I'll try again with rooms that I haven't joined yet on dev.

**Comment by srikarparsi** [comment]
just figured out a different threads issue I was working on so this is my priority now :)

**Comment by srikarparsi** [comment]
So I just got around to this and I'm not able to reproduce it. There's been a good amount of work done on joining/leaving threads and rooms so I believe something fixed it. I ran through this flow with public, workspace and private rooms:

1. User A creates a room (tried public, workspace and private)
2. User A threads a message in the room
3. User A invites User B to the room
4. User B accesses the room
5. The thread should not appear in their LHN

https://github.com/Expensify/App/assets/48188732/86d894cf-bbbc-41c4-a04b-c03c97684e50

@iwiznia I believe you reported the issue for the offshore slack channel. Is there anything I forgot to try to reproduce? Otherwise I'll close this out :)
